bundles/org.openhab.binding.anthem/src/main/java/org/openhab/binding/anthem/internal/handler/AnthemHandler.java

```diff
@@ -0,0 +1,440 @@
+import org.openhab.binding.anthem.internal.AnthemConfiguration;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.IncreaseDecreaseType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AnthemHandler} is responsible for handling commands, which are
+ * sent to one of the channels. It also manages the connection to the AV processor.
+ * The reader thread receives solicited and unsolicited commands from the processor.
+ * The sender thread is used to send commands to the processor.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class AnthemHandler extends BaseThingHandler {
+    private Logger logger = LoggerFactory.getLogger(AnthemHandler.class);
+
+    private static final long POLLING_INTERVAL_SECONDS = 900L;
+    private static final long POLLING_DELAY_SECONDS = 10L;
+
+    private @Nullable Socket socket;
+    private @Nullable BufferedWriter writer;
+    private @Nullable BufferedReader reader;
+
+    private AnthemCommandParser messageParser;
+
+    private final BlockingQueue<AnthemCommand> sendQueue = new LinkedBlockingQueue<>();
+
+    private @Nullable Future<?> asyncInitializeTask;
+    private @Nullable ScheduledFuture<?> connectRetryJob;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private @Nullable Thread senderThread;
+    private @Nullable Thread readerThread;
+
+    private int reconnectIntervalMinutes;
+    private int commandDelayMsec;
+
+    private boolean zone1PreviousPowerState;
+    private boolean zone2PreviousPowerState;
+
+    public AnthemHandler(Thing thing) {
+        super(thing);
+        messageParser = new AnthemCommandParser(this);
+    }
+
+    @Override
+    public void initialize() {
+        AnthemConfiguration configuration = getConfig().as(AnthemConfiguration.class);
+        logger.debug("AnthemHandler: Configuration of thing {} is {}", thing.getUID().getId(), configuration);
+
+        if (!configuration.isValid()) {
+            logger.debug("AnthemHandler: Config of thing '{}' is invalid", thing.getUID().getId());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/thing-status-detail-invalidconfig");
+            return;
+        }
+        reconnectIntervalMinutes = configuration.reconnectIntervalMinutes;
+        commandDelayMsec = configuration.commandDelayMsec;
+        asyncInitializeTask = scheduler.submit(this::connect);
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "@text/thing-status-detail-connecting");
```

Comment:
I missed this previously: This should be done before starting the connect task, and with status **UNKNOWN**.

Reply:
Ok